quickstart_template/301-jenkins-aptly-spinnaker-vmss.sh

```diff
@@ -159,10 +158,46 @@ run_util_script "jenkins/init-aptly-repo.sh" -vf "${vm_fqdn}" -rn "${repository_
 run_util_script "jenkins/add-aptly-build-job.sh" -al "${artifacts_location}" -st "${artifacts_location_sas_token}"
 
 echo "Setting up initial user..."
-echo "jenkins.model.Jenkins.instance.securityRealm.createAccount(\"$jenkins_username\", \"$jenkins_password\")"  > addUser.groovy
+
+# Using double quote for username and password would fail if contains dollar sign
+echo "jenkins.model.Jenkins.instance.securityRealm.createAccount('$jenkins_username', '$jenkins_password')"  > addUser.groovy
 run_util_script "jenkins/run-cli-command.sh" -cif "addUser.groovy" -c "groovy ="
 rm "addUser.groovy"
 
+# Change the Jenkins port in order not to conflict with the Spinnaker front50 port
+port=8082
+sed -i -e "s/\(HTTP_PORT=\).*/\1$port/"  /etc/default/jenkins
+service jenkins restart
+
+# If redis is not started, start and wait until port 6379 is available
+netstat -tln | grep ":6379 "
+
+while [ $? -eq 1 ]
```

Comment:
This is an infinite loop? suggest to make it finite otherwise once the service never starts, the script will hang and the usage experience will be annoying

Reply:
Change the loop to double check here